lib/src/async_cache.dart

```diff
@@ -60,12 +63,8 @@ class AsyncCache<T> {
     if (_cachedStreamSplitter != null) {
       throw StateError('Previously used to cache via `fetchStream`');
     }
-    final result = _cachedValueFuture ??= callback();
-    try {
-      return await result;
-    } finally {
-      _startStaleTimer();
-    }
+    return _cachedValueFuture ??= callback()
+      ..whenComplete(_startStaleTimer).ignore();
```

Comment:
Is this behavior change intentional? Do we still want to call `_startStaleTimer` if `_cachedValueFuture` was non-null?
```suggestion
    return (_cachedValueFuture ??= callback())
      ..whenComplete(_startStaleTimer).ignore();
```

Reply:
I think this is actually correct, and the original wasn't. The original could trigger multiple invalidations (at the same time, `_duration` after the future completed, so probably not a problem). One is sufficient.